### DISCUSSION TITLE
Request Enhancement: kmeans

### DISCUSSION THREAD

**Discussion by fmw42** [ImageMagick/ImageMagick#4857] [Development] [answered]
I would like to be able to list the final colors from -kmeans to the terminal.  

So a -define kmeans:verbose=true  or -define kmeans:colors=true would be useful

I do not see anything for this in the list of defines. So I assume it has not be implemented yet. If it has already been implemented, please point it out

**Comment by urban-warrior** [comment]
```
convert rose: -kmeans 5 gif:- | identify -verbose -
```

**Reply by fmw42** [reply]
It would be nice to get a list of hex colors and nothing else so that one does not have to do a lot of parsing outside Imagemagick.

A better alternate, but still cumbersome is

`magick rose: -kmeans 5 -depth 8 -unique-colors txt:`

**Comment by snibgo** [comment, marked as answer]
How about:
```
magick rose: -kmeans 5 -unique-colors -depth 8 -define ftxt:format=\H\n ftxt:
#413D33
#737D69
#913F32
#DF4140
#E5E4E6
```
Or if you want individual hex numbers, one number per channel:
```
magick rose: -kmeans 5 -unique-colors -define ftxt:format=\h\n ftxt:
#41bfd39a,#3dbab537,#3319476d
#73bef7fe,#7d77666a,#69b14682
#91ac71e1,#3ef941c3,#32346f8e
#e0182e64,#412edb8f,#404b669c
#e5f4e7d2,#e46dc598,#e6ec879f
```

**Reply by fmw42** [reply]
snibgo, is ftxt: included yet in Imagemagick. If so, where is the documentation for it?

**Reply by fmw42** [reply]
OK. I found it as a link to your web site from https://imagemagick.org/script/formats.php#supported
